### PR TITLE
internal: print total size of source code in analysis-stats

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: crates/rust-analyzer/src/cli/analysis_stats.rs [hunk 1]
```diff
@@ -137,6 +137,21 @@ impl flags::AnalysisStats {
...
+                total_file_size += syntax_len(db.parse(e.key).syntax_node())
+            }
+
+            let mut total_macro_file_size = Bytes::default();
+            for e in hir::db::ParseMacroExpansionQuery.in_db(db).entries::<Vec<_>>() {
+                if let Some((val, _)) = db.parse_macro_expansion(e.key).value {
+                    total_macro_file_size += syntax_len(val.syntax_node())
+                }
+            }
+            eprintln!("source files: {}, macro files: {}", total_file_size, total_macro_file_size,);
```
Comment (lnicola) [inline crates/rust-analyzer/src/cli/analysis_stats.rs:152]:
```suggestion
            eprintln!("source files: {}, macro files: {}", total_file_size, total_macro_file_size);
```

File: crates/rust-analyzer/src/cli/analysis_stats.rs [hunk 2]
```diff
@@ -361,3 +376,9 @@ fn shuffle<T>(rng: &mut Rand32, slice: &mut [T]) {
 fn percentage(n: u64, total: u64) -> u64 {
     (n * 100).checked_div(total).unwrap_or(100)
 }
+
+fn syntax_len(node: SyntaxNode) -> usize {
```
Comment (lnicola) [inline crates/rust-analyzer/src/cli/analysis_stats.rs:380]: Maybe print both to not underestimate the non-macro-generated code?

Comment (matklad) [reply crates/rust-analyzer/src/cli/analysis_stats.rs:380]: I feel that ws should mostly be free, and there's a value in printing fewer numbers

**Pull request merged**